### PR TITLE
Rename ios flag

### PR DESCRIPTION
**Pull request by 1duo**

### CODE REVIEW

File: tfcoreml/_tf_coreml_converter.py [hunk 1]
```diff
@@ -642,30 +642,30 @@ def convert(tf_model_path,
...
-  target_ios: str
-      Target Deployment iOS Version (default: '12')
-      Supported iOS version options: '11.2', '12', '13'        
-      CoreML model produced by the converter will be compatible with the iOS version specified in this argument.
-      e.g. if target_ios = '12', the converter would only utilize CoreML features released till iOS12 (equivalently macOS 10.14, watchOS 5 etc).
+  min_target_ios: str
+      Minimum target deployment iOS version (default: '12')
+      Supported iOS version options: '11.2', '12', '13'
+      Core ML model produced by the converter will be compatible with the iOS version specified in this argument.
+      e.g. if min_target_ios='12', the converter would only utilize features released till iOS 12 (equivalently macOS 10.14, watchOS 5 etc).
```
Comment (aseemw) [inline tfcoreml/_tf_coreml_converter.py]: can we say here that `Core ML model produced by the converter will be compatible with the iOS version specified in this argument and the versions after it`. e.g.: if min_target_ios='12', the converter would only utilize layers released till iOS 12 (equivalently macOS 10.14, watchOS 5 etc), and the produced model can be deployed to iOS12+ (both iOS 12 and iOS 13)

**Comment by aseemw** [comment]
I am wondering if it should be instead called `minimum_ios_deployement_target`  which is a bit longer but it matches with the setting Xcode has.

**Comment by aseemw** [comment]
The readme also needs to be updated in this PR

**Comment by aseemw** [comment]
This might be good for another PR, but the comments [here](https://github.com/tf-coreml/tf-coreml/blob/674c30572867cd9d00dc930c0ee625f5b27de757/tfcoreml/_tf_coreml_converter.py#L554) should be updated to reflect the comments [here](https://github.com/apple/coremltools/blob/875abd9707dbe65eb92a31dbb54a68d6581e68ad/coremltools/converters/tensorflow/_tf_converter.py#L34)

**Comment by 1duo** [comment]
> Lets rename the argument to `minimum_ios_deployement_target` instead, which is a bit longer but it matches with the setting Xcode has.

`minimum_ios_deployment_target` or `minimum_deployment_target_ios`?

**Comment by 1duo** [comment]
> It might be better suited for another PR, but the comments [here](https://github.com/tf-coreml/tf-coreml/blob/674c30572867cd9d00dc930c0ee625f5b27de757/tfcoreml/_tf_coreml_converter.py#L554) should be updated to reflect the comments [here](https://github.com/apple/coremltools/blob/875abd9707dbe65eb92a31dbb54a68d6581e68ad/coremltools/converters/tensorflow/_tf_converter.py#L34)

Good point!

**Comment by aseemw** [comment]
> > Lets rename the argument to `minimum_ios_deployement_target` instead, which is a bit longer but it matches with the setting Xcode has.
> 
> `minimum_ios_deployment_target` or `minimum_deployment_target_ios`?

the first one

**Comment by 1duo** [comment]
> > > Lets rename the argument to `minimum_ios_deployement_target` instead, which is a bit longer but it matches with the setting Xcode has.
> > 
> > 
> > `minimum_ios_deployment_target` or `minimum_deployment_target_ios`?
> 
> the first one

how about `minimum deployment target` and the value can be `ios12` or just `12`. That matches Xcode (Xcode has minimum deployment target).

**Comment by aseemw** [comment]
> > > > Lets rename the argument to `minimum_ios_deployement_target` instead, which is a bit longer but it matches with the setting Xcode has.
> > > 
> > > 
> > > `minimum_ios_deployment_target` or `minimum_deployment_target_ios`?
> > 
> > 
> > the first one
> 
> how about `minimum deployment target` and the value can be `ios12` or just `12`. That matches Xcode (Xcode has minimum deployment target).

yeah we can do that, although that increases our error checking surface for the string. And that would also mean that we should be able to take in strings like 'macos14', 'tvos..', 'watchos6' etc

**Comment by 1duo** [comment]
> > > > > Lets rename the argument to `minimum_ios_deployement_target` instead, which is a bit longer but it matches with the setting Xcode has.
> > > > 
> > > > 
> > > > `minimum_ios_deployment_target` or `minimum_deployment_target_ios`?
> > > 
> > > 
> > > the first one
> > 
> > 
> > how about `minimum deployment target` and the value can be `ios12` or just `12`. That matches Xcode (Xcode has minimum deployment target).
> 
> yeah we can do that, although that increases our error checking surface for that string. And that would also mean that we should be able to take in strings like 'macos14', 'tvos..', 'watchos6' etc

Yes, in that case, we'd want to handle all cases. I'm okay with either way.

**Pull request merged**